managed/services/victoriametrics/prometheus.go

```diff
@@ -209,9 +217,42 @@ func AddInternalServicesToScrape(cfg *config.Config, s models.MetricsResolutions
 		scrapeConfigForAlertmanager(s.MR),
 		scrapeConfigForGrafana(s.MR),
 		scrapeConfigForPMMManaged(s.MR),
-		scrapeConfigForQANAPI2(s.MR))
+		scrapeConfigForQANAPI2(s.MR),
+		scrapeConfigForServerPMMAgent(s.MR))
 	// TODO Refactor to remove boolean positional parameter when Prometheus is removed
 	if dbaas {
 		cfg.ScrapeConfigs = append(cfg.ScrapeConfigs, scrapeConfigForDBaaSController(s.MR))
 	}
 }
+
+// AddPmmAgentScrapeConfig adds pmm-agent metrics to scrape targets.
+func AddPmmAgentScrapeConfig(l *logrus.Entry, q *reform.Querier, s *models.MetricsResolutions, pmmAgentID *string) (*config.ScrapeConfig, error) {
+	agent, err := models.FindAgentByID(q, pointer.GetString(pmmAgentID))
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	// find Node for this Agent
+	node, err := models.FindNodeByID(q, pointer.GetString(agent.RunsOnNodeID))
+	if err != nil {
+		return nil, err
+	}
+
+	pmmAgentVersion, err := version.Parse(pointer.GetString(agent.Version))
+	if err != nil {
+		l.Warnf("couldn't parse pmm-agent version for pmm-agent %s: %q", agent.AgentID, err)
```

Comment:
I guess we need return here

Reply:
Hm, no, probably here we can continue without return. But do we use `pmmAgentVersion` somewhere at all?